src/main/java/org/mule/extension/api/introspection/ImmutableSourceModel.java

```diff
@@ -46,13 +48,15 @@ public ImmutableSourceModel(String name,
                                 DataType attributesType,
                                 SourceFactory sourceFactory,
                                 Map<String, Object> modelProperties,
-                                List<InterceptorFactory> interceptorFactories)
+                                List<InterceptorFactory> interceptorFactories,
+                                Optional<ExceptionEnricherFactory> exceptionEnricherFactory)
```

Comment:
There is a method that fetches the EnricherFactory if exists, if not an Optional.empty() is returned, in the construction of the instance we only make a call to that method. IMO is an overkill to have 2 constructors.